spec/services/order_cycle_form_spec.rb

```diff
@@ -271,22 +342,90 @@
 
     context "updating shipping methods" do
       context "and it's valid" do
-        it "saves the changes" do
-          distributor = create(:distributor_enterprise)
-          distributor_shipping_method = create(
-            :shipping_method,
-            distributors: [distributor]
-          ).distributor_shipping_methods.first
-          order_cycle = create(:distributor_order_cycle, distributors: [distributor])
+        let(:distributor){ create(:distributor_enterprise) }
+        let(:shipping_method){ create(:shipping_method, distributors: [distributor]) }
+        let(:shipping_method2){ create(:shipping_method, distributors: [distributor]) }
+        let(:distributor_shipping_method){ shipping_method.distributor_shipping_methods.first }
+        let(:distributor_shipping_method2){ shipping_method2.distributor_shipping_methods.first }
+        let(:supplier){ create(:supplier_enterprise) }
+        context "the submitter is a coordinator" do
+          it "saves the changes" do
+            order_cycle = create(:distributor_order_cycle, distributors: [distributor])
+
+            form = OrderCycleForm.new(
+              order_cycle,
+              { selected_distributor_shipping_method_ids: [distributor_shipping_method.id] },
+              order_cycle.coordinator.users.first
+            )
 
-          form = OrderCycleForm.new(
-            order_cycle,
-            { selected_distributor_shipping_method_ids: [distributor_shipping_method.id] },
-            order_cycle.coordinator
-          )
+            expect(form.save).to be true
+            expect(order_cycle.distributor_shipping_methods).to eq [distributor_shipping_method]
+          end
+        end
+        context "submitter is a supplier" do
+          it "doesn't save the changes" do
```

Comment:
I just tried to implement a `change` block, but can't get it to fail. `order_cycle.distributor_shipping_methods` seems to be changing after `OrderCycleForm.new`, which is weird, because it shouldn't change until `form.save`. 

I'm obviously not seeing things right so I'll hand back to you.

Reply:
This is weird. There could be another problem that we overlooked. 
Can you please share the code?

Reply:
I can see it's updated with `form.save`.  I checked out the latest code and figured out how to make it fail: I made `can_update_selected_payment_or_shipping_methods?` return `true` and it fails, which seems right.

All good!